packages/flutter_tools/lib/src/hot.dart

```diff
@@ -176,16 +197,16 @@ class HotRunner extends ResidentRunner {
   }
 
   @override
-  void handleTerminalCommand(String code) {
+  Future<Null> handleTerminalCommand(String code) async {
     final String lower = code.toLowerCase();
     if (lower == 'r' || code == AnsiTerminal.KEY_F5) {
       // F5, restart
       if (code == 'r') {
         // lower-case 'r'
-        _reloadSources();
+        await _reloadSources();
       } else {
         // upper-case 'R'.
```

Comment:
this code is wrong. it makes F5 into a cold restart. It should be a warm restart.